### PR TITLE
Remove unused and update dependencies

### PR DESCRIPTION
**Pull request by FDiskas**
Clean some unused stuff

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -1,28 +1,26 @@
 module.exports = {
-  parser: 'babel-eslint',
-  extends: ['standard', 'standard-react', 'prettier', 'react-app'],
-  plugins: ['prettier', 'react'],
+  env: {
+    browser: true,
+    es6: true,
+    node: true,
+  },
+  extends: ['plugin:react/recommended', 'airbnb', 'prettier'],
```
Comment (yamalight) [inline .eslintrc.js]: I'd prefer using standard over airbnb tbh.

File: package.json [hunk 2]
```diff
@@ -18,56 +32,25 @@
     "loaders.css": "^0.1.2",
     "lodash": "^4.17.15",
     "node-fetch": "^2.6.0",
-    "prop-types": "15.7.2",
```
Comment (yamalight) [inline package.json]: I'd keep this. It'd be good to add proper PropTypes descriptions at one point

File: package.json [hunk 3]
```diff
@@ -18,56 +32,25 @@
...
-  ],
-  "license": "MIT",
-  "scripts": {
-    "analyze:lint": "eslint --ext .jsx --ext .js .",
-    "analyze:prettier": "prettier --list-different '**/*.{js,css,md,json}'",
-    "analyze": "npm run analyze:lint && npm run analyze:prettier",
-    "build": "gatsby build",
-    "develop": "gatsby develop",
-    "fix:lint": "eslint --ext .jsx --ext .js . --fix",
-    "fix:prettier": "prettier --write '**/*.{js,css,md,json}'",
```
Comment (yamalight) [inline package.json:41]: Why did you decide to remove `fix` command?

File: plugins/gatsby-source-bxjs/episodesToDocuments.js [hunk 4]
```diff
@@ -19,12 +20,12 @@ exports.markdownToDocuments = text => {
     .map(section => section.replace(/\r/g, ''))
     .filter(
       section =>
-        section && section.length > 0 && section.replace(/\n/g, '').length > 0
+        section && section.length > 0 && section.replace(/\n/g, '').length > 0,
```
Comment (yamalight) [inline plugins/gatsby-source-bxjs/episodesToDocuments.js]: I prefer not having trailing commas in functions

File: plugins/gatsby-source-bxjs/episodesToDocuments.js [hunk 5]
```diff
@@ -19,12 +20,12 @@ exports.markdownToDocuments = text => {
...
     )
-    .flatMap(text => {
-      const [name, linksText] = text.split(/:\n/g);
+    .flatMap(text2 => {
```
Comment (yamalight) [inline plugins/gatsby-source-bxjs/episodesToDocuments.js]: I get that it shadows the var in higher scope, but `text2` is a terrible var name 😅

File: plugins/gatsby-source-bxjs/gatsby-node.js [hunk 6]
```diff
@@ -68,7 +72,7 @@ exports.sourceNodes = async ({ actions }) => {
 
     // create new episode and use links inside of it
     const newEpisode = {
-      id: `${docId++}`,
+      id: `${(docId += 1)}`,
```
Comment (yamalight) [inline plugins/gatsby-source-bxjs/gatsby-node.js]: Now this is just silly, there's no reason to not use `++`

**Review by yamalight — CHANGES_REQUESTED** [review]
Thanks for the PR, but that's way too many changes that I don't feel are necessary 🤔

File: .eslintrc.js (hunk 1 above)
Comment (FDiskas) [reply .eslintrc.js]: But you wasn't. Standard is more simplified. I changed to airbnb and linter not claiming to much

File: package.json (hunk 3 above)
Comment (FDiskas) [reply package.json:41]: I left only `lint` command with `--fix`

File: .eslintrc.js (hunk 1 above)
Comment (FDiskas) [reply .eslintrc.js]: But yes - probably it's a good time to update this

Comment (yamalight) [reply .eslintrc.js]: What do you mean I wasn't? It was right there in eslint before you replaced it with airbnb 🤔 
Airbnb's config has way too much silly rules IMO

File: package.json (hunk 3 above)
Comment (yamalight) [reply package.json:41]: Those should be two different commands. One is executed to fix stuff, while other is executed in CI to make sure code is formatted correctly

File: plugins/gatsby-source-bxjs/episodesToDocuments.js (hunk 5 above)
Comment (FDiskas) [reply plugins/gatsby-source-bxjs/episodesToDocuments.js]: Actually airbnb lint rules showed that error, but standard ruleset are happy. Should I revert it back?

File: .eslintrc.js (hunk 1 above)
Comment (FDiskas) [reply .eslintrc.js]: There is a good point https://github.com/BuildingXwithJS/BxJS-Website/pull/33#discussion_r398907002

Comment (FDiskas) [reply .eslintrc.js]: reverted

**Comment by FDiskas** [comment]
Also updated some packages to remove vulnerabilities 
```
npm audit
```

File: .eslintrc.js [hunk 7]
```diff
@@ -1,28 +1,36 @@
 module.exports = {
-  parser: 'babel-eslint',
-  extends: ['standard', 'standard-react', 'prettier', 'react-app'],
```
Comment (yamalight) [inline .eslintrc.js:3]: why not just leave it as is? why do you want to change them? 🤔

File: .eslintrc.js [hunk 8]
```diff
@@ -1,28 +1,36 @@
...
+    },
+    ecmaVersion: 2018,
+    sourceType: 'module',
+  },
+  plugins: ['react', 'prettier'],
   rules: {
-    camelcase: 'off',
+    'prettier/prettier': 'error',
     'react/prop-types': 'off',
-    'react/react-in-jsx-scope': 'off',
```
Comment (yamalight) [inline .eslintrc.js:8]: Same goes for this rules - they are here for a reason 🤔

File: gatsby-node.js [hunk 9]
```diff
@@ -20,4 +20,6 @@ exports.createPages = async ({ actions: { createPage }, graphql }) => {
       context: {},
     });
   });
+
+  return Promise.resolve();
```
Comment (yamalight) [inline gatsby-node.js:24]: This is redundant

File: package.json [hunk 10]
```diff
@@ -24,50 +39,23 @@
...
   },
-  "keywords": [
-    "gatsby",
-    "tailwind",
-    "tailwindcss",
-    "purgecss"
-  ],
-  "license": "MIT",
-  "scripts": {
-    "analyze:lint": "eslint --ext .jsx --ext .js .",
```
Comment (yamalight) [inline package.json:35]: what's the reasoning behind removing / changing all existing scripts?

File: plugins/gatsby-source-bxjs/package.json [hunk 11]
```diff
@@ -7,5 +7,11 @@
   },
   "keywords": [],
   "author": "",
-  "license": "MIT"
+  "license": "MIT",
+  "dependencies": {
```
Comment (yamalight) [inline plugins/gatsby-source-bxjs/package.json:11]: those deps are installed and used from top level, not really needed here

File: src/components/layout.js [hunk 12]
```diff
@@ -1,5 +1,4 @@
 import 'loaders.css';
-import PropTypes from 'prop-types';
```
Comment (yamalight) [inline src/components/layout.js:2]: My old comment on keeping this as-is stands.

**Review by yamalight — CHANGES_REQUESTED** [review]
I'm still not sure why you want to do so many changes to eslint config and existing scripts.
Could you please explain your reasoning here?

File: .eslintrc.js (hunk 7 above)
Comment (FDiskas) [reply .eslintrc.js:3]: They where not working. Eslint was not shoing prettier errors. Some of the dependencies got some vulnerabilities.

File: .eslintrc.js (hunk 8 above)
Comment (FDiskas) [reply .eslintrc.js:8]: Redundant

File: package.json (hunk 10 above)
Comment (FDiskas) [reply package.json:35]: Thous scripts was not used anywhere. For example this one does the same lint command. Other prittier related scripts also. Prittier now integrated with eslint

File: plugins/gatsby-source-bxjs/package.json (hunk 11 above)
Comment (FDiskas) [reply plugins/gatsby-source-bxjs/package.json:11]: Let it be independent

Comment (FDiskas) [reply plugins/gatsby-source-bxjs/package.json:11]: yarn installs them anyway. So later if you drop that plugin - no left owers

File: gatsby-node.js (hunk 9 above)
Comment (FDiskas) [reply gatsby-node.js:24]: Realy? the function is async and in if statement it returns a promise.reject() but in other case it does not returns anything.

File: src/components/layout.js (hunk 12 above)
Comment (FDiskas) [reply src/components/layout.js:2]: ```children: PropTypes.node.isRequired,``` is redundant - so nothing is left. Why let this import to be here?

Comment (yamalight) [reply src/components/layout.js:2]: As I've said before - I do want to have proper proptype definitions in the long run

File: plugins/gatsby-source-bxjs/package.json (hunk 11 above)
Comment (yamalight) [reply plugins/gatsby-source-bxjs/package.json:11]: But we can't really drop it though

File: package.json (hunk 10 above)
Comment (yamalight) [reply package.json:35]: They are there for convenience. Scripts don't have to be used immediately

File: gatsby-node.js (hunk 9 above)
Comment (yamalight) [reply gatsby-node.js:24]: async function always returns a promise.
`Promise.reject()` is used to basically throw an error (that's what I should've done in a first place 🤔 )

File: .eslintrc.js (hunk 7 above)
Comment (yamalight) [reply .eslintrc.js:3]: I am pretty sure it was working fine 🤔

**Pull request closed without merging**